src/domain/usecases/UpdateEventDataValueUseCase.ts

```diff
@@ -0,0 +1,104 @@
+import _ from "lodash";
+import { createObjectCsvWriter } from "csv-writer";
+import { Async } from "domain/entities/Async";
+import { Id } from "domain/entities/Base";
+import { EventMetadata, Stats } from "domain/entities/Event";
+import { EventRepository } from "domain/repositories/EventRepository";
+import logger from "utils/log";
+
+export type MigrateOptions = {
+    eventId: Id[];
+    dataElementId: Id;
+    condition: string;
+    newValue: string;
+    csvPath: string;
+    post: boolean;
+};
+
+export class UpdateEventDataValueUseCase {
+    constructor(private eventRepository: EventRepository) {}
+
+    async execute(options: MigrateOptions): Async<Stats> {
+        const eventMetadata = await this.eventRepository.getByIds(options.eventId);
+        const eventsWithDvInCondition = this.getEventsInCondition(eventMetadata, options);
+
+        if (options.csvPath) {
+            logger.debug(`Generate report: ${options.csvPath}`);
+            await this.generateCsvReport(eventMetadata, options);
+        }
+
+        if (options.post) {
+            logger.debug(`Events to change: ${eventsWithDvInCondition.length}`);
+            const stats = await this.eventRepository.saveAll(eventsWithDvInCondition);
+            return stats;
+        }
+
+        return {
+            created: 0,
+            ignored: 0,
+            updated: 0,
+        };
```

Comment:
As a rule of thumb, favour full if/else (possible exception: when the if acts as guard). if (options.post) { ... } else { ... }